File: Assignment__18/Assignment_1.c
#include<stdio.h>
#include<stdlib.h>

int DifferenceEvenOdd(int Arr[] , int iLength)
{
    int iSum1 = 0, iSum2 = 0, iCnt = 0 , Diff = 0 ;
    for(iCnt = 0 ; iCnt < iLength;iCnt++)
    {
        if(Arr[iCnt] % 2 == 0)
        {
            iSum1 = iSum1 + Arr[iCnt];
        }
        else 
        {
            iSum2 = iSum2 + Arr[iCnt];
        }
    }
    Diff = iSum1 - iSum2;
    printf("\t(%d - %d)\n",iSum1,iSum2);
    return Diff;

}
int main()
{
    int iSize = 0 , iCnt = 0 ;
    int iRet = 0 ;
    int *ptr = NULL;

    printf("Enter number of Elements :  ");
    scanf("%d",&iSize);

    ptr = (int *)malloc(iSize * sizeof(int));
    if(ptr == NULL)
    {
        printf("Unable to locate memory : \n");
        return -1;
    }

    printf("Enter %d numbers : \n",iSize);
    for(iCnt = 0  ; iCnt < iSize ; iCnt++)
    {
        printf("Enter %d numbers : ",iCnt+1);
        scanf("%d",&ptr[iCnt]);
    }

    iRet = DifferenceEvenOdd(ptr , iSize); 
    printf("Difference between Even Odd is : %d ",iRet);

    free(ptr);

    return 0 ;
}

File: Assignment__18/Assignment_java/Assignment_1.java


import java.util.*;

class Difference
{
    public int SumEvenOdd(int [] Arr , int iLength)
    {
        int iSumEven = 0 , iSumOdd = 0, iDiff = 0 ;
        for(int iCnt = 0 ; iCnt < iLength ; iCnt++)
        {
            if(Arr[iCnt] % 2  == 0 )
            {
                iSumEven = iSumEven + Arr[iCnt];
            }
            else 
            {
                iSumOdd = iSumOdd + Arr[iCnt];
            }
        }
        iDiff = iSumEven - iSumOdd;
        System.out.println("("+iSumEven+" - "+iSumOdd+")");
        return iDiff;
    }
}
class Assignment_1 
{

    public static void main(String A [])
    {
        Scanner sobj = new Scanner(System.in);
       

        System.out.println("Enter number  : ");
        int iSize = sobj.nextInt();
        int []ptr = new int[iSize];

        System.out.println("Enter Elements  : "+iSize);

        for(int iCnt = 0 ; iCnt < iSize; iCnt++)
        {
            System.out.print("Enter Elements : "+(iCnt + 1)+" : ");
            ptr[iCnt] = sobj.nextInt();

        }
        Difference dobj = new Difference();
        int iRet = dobj.SumEvenOdd(ptr ,iSize);
        System.out.println("Result is : "+iRet);
    }
}